### PR TITLE
[AC-2323] Flexible collections: automatically migrate data for all Organizations

### PR DESCRIPTION
**Pull request by r-tome**
## Type of change

```
- [ ] Bug fix
- [X] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Migrate all organizations data to use enhanced collections. Override `FlexibleCollectionsSignup` feature flag to true for local usage.

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: util/Migrator/DbScripts/2024-03-25_00_EnableAllOrgCollectionEnhancements.sql
Comment (rkac-bw) [inline util/Migrator/DbScripts/2024-03-25_00_EnableAllOrgCollectionEnhancements.sql:3]: Don't think this comment applies as you are not doing a percentage

Comment (r-tome) [reply util/Migrator/DbScripts/2024-03-25_00_EnableAllOrgCollectionEnhancements.sql]: Fixed it!

**Review by rkac-bw — APPROVED** [review]
LGTM

**Review by eliykat — COMMENTED** [review]
Lgtm, please pass this to QA to start testing the mssql migration.

I will hold off properly approving until we've merged in the EF migrations and it's all ready to go. We also need to ensure all the cloud migrations are complete and it's passed QA.

EDIT: The enforce labels workflow doesn't seem to be blocking merge so I've dismissed the approving review just to make sure.

**Review by rkac-bw — APPROVED** [review]
LGTM

**Comment by r-tome** [comment]
@rkac-bw @eliykat I've bumped the dates on the migration scripts so its ready for merging

**Review by eliykat — APPROVED** [review]
:shipit:

**Review by rkac-bw — APPROVED** [review]

**Pull request merged**
